core/src/main/scala/blobstore/StoreOps.scala

```diff
@@ -17,15 +17,15 @@ package blobstore
 
 import java.nio.charset.Charset
 
-import cats.effect.{Effect, Sync}
+import cats.effect.{ConcurrentEffect, ContextShift, Effect, Sync}
 import fs2.{Pipe, Sink, Stream}
 import cats.implicits._
 
 import scala.concurrent.ExecutionContext
 
 trait StoreOps {
 
-  implicit class PutOps[F[_]](store: Store[F]) {
+  implicit class PutOps[F[_] : ContextShift](store: Store[F])(implicit ec: ExecutionContext) {
```

Comment:
Is it possible to move the constraints to the functions themselves? This makes the class more general